src/parser/SparqlParser.cpp

```diff
@@ -806,21 +806,21 @@ void SparqlParser::addLangFilter(
     const std::string& lhs, const std::string& rhs,
     std::shared_ptr<ParsedQuery::GraphPattern> pattern) {
   auto langTag = rhs.substr(1, rhs.size() - 2);
-  // first find a predicate for the given variable
-  auto it = std::find_if(pattern->_whereClauseTriples.begin(),
-                         pattern->_whereClauseTriples.end(),
-                         [&lhs](const auto& tr) { return tr._o == lhs; });
-  while (it != pattern->_whereClauseTriples.end() &&
-         it->_p._operation == PropertyPath::Operation::IRI &&
-         ad_utility::startsWith(it->_p._iri, "?")) {
-    it = std::find_if(it + 1, pattern->_whereClauseTriples.end(),
-                      [&lhs](const auto& tr) { return tr._o == lhs; });
-  }
+  // First find a suitabke triple for the given variable. It
+  // must use a predicate that is not a variable or complex
```

Comment:
Do non-complex predicate paths exist?

Reply:
Yes because everything is a predicate path now so just a predicate is a non-complex path